### PR TITLE
Add Gauss-Chebyshev quadrature

### LINKED ISSUE #51: GaussJacobi panics for some valid inputs
`GaussJacobi::new(10, -0.5, -0.5)` panics because the nodes and weights are evaluated to `NaN`, and then the comparison panics in the call to `sort_unstable_by`. I have added a test that finds this bug in the branch `found_nans`.

### PR DESCRIPTION
**Pull request by JSorngard**
This PR adds Gauss-Chebyshev quadrature of the first and second kind inspired by the conversation in #6.  
It also implements `From<GaussChebyshev>` and `From<GaussLegendre>` for `GaussJacobi` and delegates to those constructors when `alpha` and `beta` have appropriate values. This is because it partially resolve #51, as `GaussJacobi::new` no longer panics for the inputs that are now computed with Gauss-Chebyshev quadrature instead.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,31 +1,37 @@
+# Changelog
```
Comment (JSorngard) [inline CHANGELOG.md:1]: Most of the changes to this file came along when I linted markdown files without thinking. They do not need to be part of this PR and could be reverted.

**Comment by JSorngard** [comment]
I still need to add tests to the `chebyshev` module.

**Comment by JSorngard** [comment]
I also want to add some more tests on an interval other than [-1,1].

**Comment by JSorngard** [comment]
What do you think of this implementation and API @DomiDre?

**Review by DomiDre — COMMENTED** [review]
I think it's very nice. We should also push the version in the Cargo.toml and then I think it's good to merge and to push a new release from my point of view. Nice work!

**Comment by JSorngard** [comment]
Good point, done!

**Pull request merged**